var/spack/repos/builtin/packages/dla-future/package.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class DlaFuture(CMakePackage, CudaPackage, ROCmPackage):
+    """DLA-Future library: Distributed Linear Algebra with Future"""
+
+    homepage = "https://github.com/eth-cscs/DLA-Future"
+    url = "https://github.com/eth-cscs/DLA-Future/archive/v0.0.0.tar.gz"
+    git = "https://github.com/eth-cscs/DLA-Future.git"
+    maintainers = ["rasolca", "albestro", "msimberg", "aurianer"]
+
+    version("0.1.0", sha256="f7ffcde22edabb3dc24a624e2888f98829ee526da384cd752b2b271c731ca9b1")
+    version("develop", branch="master")
+
+    cxxstds = ("17", "20")
+    variant(
+        "cxxstd",
+        default="17",
+        values=cxxstds,
+        description="Use the specified C++ standard when building",
+    )
+    conflicts("cxxstd=20", when="+cuda")
+
+    variant("shared", default=True, description="Build shared libraries.")
+
+    variant("doc", default=False, description="Build documentation.")
+
+    variant("miniapps", default=False, description="Build miniapps.")
+
+    depends_on("cmake@3.22:", type="build")
+    depends_on("doxygen", type="build", when="+doc")
+    depends_on("mpi")
+    depends_on("blaspp@2022.05.00:")
+    depends_on("lapackpp@2022.05.00:")
+
+    depends_on("umpire~examples")
+    depends_on("umpire+cuda~shared", when="+cuda")
+    depends_on("umpire+rocm~shared", when="+rocm")
+    depends_on("umpire@4.1.0:")
+
+    depends_on("pika@0.15.1:")
+    depends_on("pika-algorithms@0.1:")
+    depends_on("pika +mpi")
+    depends_on("pika +cuda", when="+cuda")
+    depends_on("pika +rocm", when="+rocm")
+    for cxxstd in cxxstds:
+        depends_on("pika cxxstd={0}".format(cxxstd), when="cxxstd={0}".format(cxxstd))
+        depends_on("pika-algorithms cxxstd={0}".format(cxxstd), when="cxxstd={0}".format(cxxstd))
+
+    for build_type in ("Debug", "RelWithDebInfo", "Release"):
+        depends_on(
+            "pika build_type={0}".format(build_type), when="build_type={0}".format(build_type)
+        )
+        depends_on(
+            "pika-algorithms build_type={0}".format(build_type),
+            when="build_type={0}".format(build_type),
+        )
```

Comment:
Do we still want to enforce this? This is not a requirement by `pika` or `pika-algorithms` anymore (at least I think the constraint was lifted before `pika` 0.1.0). Without it one can still do `dla-future build_type==Debug` which is a bit more aggressive than the manual constraint added here. I'd like to avoid overconstraining the variants.

Reply:
Removed in https://github.com/spack/spack/pull/38224/commits/f700b79c205361607915ffbd7e2a911117635134